nifi-nar-bundles/nifi-geohash-bundle/nifi-geohash-processors/src/main/java/org/apache/nifi/processors/geohash/GeohashRecord.java

```diff
@@ -0,0 +1,449 @@
+            }
+
+            final WriteResult writeResult = writer.finishRecordSet();
+            writer.close();
+
+            WriteResult notFoundWriterResult = null;
+
+            if (notFoundWriter != null) {
+                notFoundWriterResult = notFoundWriter.finishRecordSet();
+                notFoundWriter.close();
+            }
+
+            is.close();
+            os.close();
+
+            if (osNotFound != null) {
+                osNotFound.close();
+            }
+
+            output = session.putAllAttributes(output, buildAttributes(foundCount, writer.getMimeType(), writeResult));
+            if (routingStrategy != RoutingStrategy.SPLIT) {
+                session.transfer(output, targetRelationship);
+                session.remove(input);
+            } else {
+                if (notFoundCount > 0) {
+                    notFound = session.putAllAttributes(notFound, buildAttributes(notFoundCount, writer.getMimeType(), notFoundWriterResult));
+                    session.transfer(notFound, REL_FAILURE);
+                } else {
+                    session.remove(notFound);
+                }
+                session.transfer(output, REL_SUCCESS);
+                session.transfer(input, REL_ORIGINAL);
+            }
+
+        } catch (Exception ex) {
+            getLogger().error("Failed to {} due to {}", encode ? "encode" : "decode", ex.getLocalizedMessage(), ex);
+        }
+    }
+
+
+    private Object getEncodedGeohash(RecordPath latitudePath, RecordPath longitudePath, Record record, String format, int level) {
+        RecordPathResult latitudeResult = latitudePath.evaluate(record);
+        RecordPathResult longitudeResult = longitudePath.evaluate(record);
+        Optional<FieldValue> latitudeField = latitudeResult.getSelectedFields().findFirst();
+        Optional<FieldValue> longitudeField = longitudeResult.getSelectedFields().findFirst();
+
+        if (!latitudeField.isPresent() || !longitudeField.isPresent()) {
+            return null;
+        }
+
+        FieldValue latitudeValue = latitudeField.get();
+        FieldValue longitudeValue = longitudeField.get();
+        Object latitudeVal = latitudeValue.getValue();
+        Object longitudeVal = longitudeValue.getValue();
+
+        if (latitudeVal == null || longitudeVal == null) {
+            return null;
+        }
+
+        double realLatValue = Double.parseDouble(latitudeVal.toString());
+        double realLongValue = Double.parseDouble(longitudeVal.toString());
+        GeoHash gh = GeoHash.withCharacterPrecision(realLatValue, realLongValue, level);
+
+        switch (GeohashFormat.valueOf(format)) {
+            case BINARY_STRING:
+                return gh.toBinaryString();
+            case LONG:
+                return gh.longValue();
+            default:
+                return gh.toBase32();
+        }
+    }
+
+    private WGS84Point getDecodedPointFromGeohash(RecordPath geohashPath, Record record, String format) {
```

Comment:
As mentioned above, recommend using `GeohashFormat`:
```suggestion
    private WGS84Point getDecodedPointFromGeohash(RecordPath geohashPath, Record record, GeohashFormat format) {
```